src/Illuminate/Support/Str.php

```diff
@@ -679,6 +687,38 @@ public static function isUlid($value)
         return Ulid::isValid($value);
     }
 
+    /**
+     * Determine if the given value is a valid email.
+     *
+     * @param  mixed  $value
+     * @param  array<int, int|string>  $parameters
```

Comment:
You could narrow this further down:
```suggestion
     * @param  array<int, int|'strict'|'dns'|'spoof'|'filter'|'filter_unicode'|class-string>  $parameters
```

Reply:
@shaedrich Sure, but @taylorotwell said:

> I wouldn't use parameters here like we do in validation. You can use named arguments for better readability.

Should I go for this instead?

```php
public static function isEmail(
    $value,
    bool $strict = false,
    bool $dns = false,
    bool $spoof = false,
    bool $filter = false,
    bool $filterUnicode = false,
    array $customValidations = [],
) {}
```

Or I'd love a better suggestion than my proposal.